### PR TITLE
Stripe API setup, Stripe Connect setup

### PR DESCRIPTION
**Pull request by cl-codes**
Connected to the following issues:
https://github.com/hatchways/team-oak/issues/40
https://github.com/hatchways/team-oak/issues/63

### What this PR does (required):
- Stripe libraries added.
- Set up a new Stripe account for the team. Credentials and API keys will be shared separately.
- Sample environment variables updated with placeholders for API keys, as well as redirect links used by Stripe.
- Settings > Billing currently includes a button for petsitters. When clicked, it will redirect them to setup their account on Stripe. Redirects back to our site when done.
- Profile model updated so that we store the user's Stripe ID.
- Stripe controller  and router updated on the backend as well.

### Screenshot:
Screen Shot 2022-01-21 at 12 15 19 AM

### Any information needed to test this feature:
- Sign in, go to settings > billing, click the `Connect to Stripe` button. For now, this Button is disabled after a stripe account is created on the first click (even if they haven't fully onboarded yet). 
- Complete the onboarding process as a test user (credit card and ID information can be skipped as a test user)
- Stripe provides a link to return to our site - test this too.
- Log into team oak's stripe account (test mode) to see list of users connected, and whether they have been onboarded. 

### Any issues with the current functionality (optional):
- Would be nice to preload some more information before generating stripe connect link. Stripe is strict about the data they accept, and our demo address and phone number aren't currently accepted. 
- Our profile model only tracks whether a user has a stripe ID. This does not necessarily mean that the user has completed onboarding through stripe - further work to be done on checking onboarding status, and FE work to prompt user to complete the process.
- Stripe link expires at a certain time (when we get the redirect url, the Stripe API also returns the expiry time). Should prompt petsitter to create a new link if they haven’t finished onboarding.

## Notes for the Future:
- If we ever push the app to production - remember to change the refresh and return links provided to Stripe. These are saved to the .env variable so we'll be more likely to remember.

### CODE REVIEW

File: server/controllers/stripe.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    company: {
+      name: profile.name,
+    },
+    country: 'CA',
+    default_currency: 'cad',
+    type: 'standard', 
+    });
+
+    profile.stripeAccountId = account.id;
+    await profile.save()
```
Comment (moffatethan) [inline server/controllers/stripe.js:23]: Hm - just thinking about this now. They could have their Stripe account created but their link expires. Should we check for this as well? I might make this a new ticket as I think about how to go about doing this.

File: client/src/components/ConnectStripe/ConnectStripe.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+interface StripeProps {
+  header: string;
+}
+
+export const ConnectStripe: React.FC<StripeProps> = ({ header }) => {
+  const { profile } = useAuth();
+
+  const clickHandler = async () => {
+    await getLink().then((data: StripeLink) => {
```
Comment (moffatethan) [inline client/src/components/ConnectStripe/ConnectStripe.tsx:18]: If it fails could we send a message too? Just something like "An error occured while connecting Stripe". Take a look at the useSnackbar context which we can make use of here.

**Review by moffatethan — APPROVED** [review]
Tested locally, works great! We're definitely going in the right direction and we'll iterate as needed but this is a solid start. Did a PaymentIntent test too and destination account ID was working fine. We'll need to start working on our payment system more on the back-end and front-end to really leverage Stripe.

**Comment by moffatethan** [comment]
@cl-codes Please address the merge conflict too

File: server/controllers/stripe.js (hunk 1 above)
Comment (cl-codes) [reply server/controllers/stripe.js:23]: Sounds good - will give this some more thought too!

File: client/src/components/ConnectStripe/ConnectStripe.tsx (hunk 2 above)
Comment (cl-codes) [reply client/src/components/ConnectStripe/ConnectStripe.tsx:18]: updated in latest version - thanks!

**Comment by cl-codes** [comment]
> @cl-codes Please address the merge conflict too

Merged in latest version - thanks!

**Pull request merged**